python/cuda_parallel/tests/test_reduce.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+#
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import numpy
+import pytest
+from numba import cuda
+import cuda.parallel.experimental as cudax
+
+
+def random_int(shape, dtype):
+    return numpy.random.randint(0, 5, size=shape).astype(dtype)
+
+
+def type_to_problem_sizes(dtype):
+    if dtype in [numpy.uint8, numpy.int8]:
+        return [2, 4, 5, 6]
+    elif dtype in [numpy.uint16, numpy.int16]:
+        return [4, 8, 12, 14]
+    elif dtype in [numpy.uint32, numpy.int32]:
+        return [16, 20, 24, 28]
+    elif dtype in [numpy.uint64, numpy.int64]:
+        return [16, 20, 24, 28]
+    else:
+        raise ValueError("Unsupported dtype")
+
+
+@pytest.mark.parametrize('dtype', [numpy.uint8, numpy.uint16, numpy.uint32, numpy.uint64])
+def test_device_reduce(dtype):
+    def op(a, b):
+        return a + b
+
+    init_value = 42
+    h_init = numpy.array([init_value], dtype=dtype)
```

Comment:
curious: Why is this an array and not a scalar?

Reply:
That stems from the need in type erasure. We can't pass user-defined types by value, so we have to take an address. There's no way to take an address of a Python variable. So we pass the initial value through numpy array that has a C++ memory representation of underlying elements and can give us an address for later kernel launch.